### DISCUSSION TITLE
Could not find the mark of the transition "Branch"

### DISCUSSION THREAD

**Discussion by dascrew** [ogamespec/ogame-opensource#127] [Q&A] [answered]
This error is getting thrown when running AI. Branch is transitioning into Label in the visual programmer, am I not using it correctly?

**Comment by ogamespec** [comment]
Have you seen this?

https://github.com/ogamespec/ogame-opensource/blob/master/Wiki/en/ai.md

**Reply by dascrew** [reply]
Yes this what I have been using, very informative! The chain executes without issue until a branch, perhaps I am missing something from the instructions. As far as I understand it the branch should chain out to a label?

**Reply by ogamespec** [reply]
Well, yes, it should work.
Another thing is that it may not be worked out the way you want it to :)
This work is still to be done (approximately after localization of the project)

**Reply by dascrew** [reply]
![branchissue](https://github.com/ogamespec/ogame-opensource/assets/166811049/8784cf2f-9249-46b7-9c0f-e62653bc8f44)
This is how I was wanting, looking forward to all the improvements coming in future releases.

**Reply by ogamespec** [reply]
Аh. That's just an example. The bottom blocks (Branch) indicate which of the top blocks (Label) you want to jump to.

**Reply by ogamespec** [reply, marked as answer]
![image](https://github.com/ogamespec/ogame-opensource/assets/5828819/25aecfe3-38e4-4392-99be-b5edeb4dc9d3)
I marked what I meant in the picture.

**Reply by ogamespec** [reply]
Such a technique is called "Common Bus" or DRAKON.
